examples/bindings/quickstart_sparse_string/Program.cs

```diff
@@ -0,0 +1,321 @@
+                data_handle.AddrOfPinnedObject().ToPointer(), (ulong*)data_size_handle.AddrOfPinnedObject()
+            );
+            data_handle.Free();
+            data_size_handle.Free();
+            check_err(ctx);
+
+            Methods.tiledb_query_submit(ctx, query);
+            check_err(ctx);
+
+            Marshal.FreeHGlobal((IntPtr)attr_name);
+            Marshal.FreeHGlobal((IntPtr)rows_name);
+            Marshal.FreeHGlobal((IntPtr)cols_name);
+            Methods.tiledb_array_close(ctx, array);
+            check_err(ctx);
+            Methods.tiledb_query_free(&query);
+        }
+
+        static unsafe void read_array(tiledb_ctx_t* ctx, tiledb_array_t* array)
+        {
+            Methods.tiledb_array_open(ctx, array, tiledb_query_type_t.TILEDB_READ);
+            check_err(ctx);
+
+            byte[] read_rows = new byte[4];
+            UInt64 read_rows_size = 4;
+            
+            UInt64[] read_rows_offsets = new UInt64[3];
+            UInt64 read_rows_offsets_size = (UInt64)read_rows_offsets.Length * sizeof(UInt64);
+            
+            Int32[] read_cols = new Int32[3];
+            UInt64 read_cols_size = (UInt64)3 * sizeof(Int32);
+            
+            Int32[] read_data = new Int32[3];
+            UInt64 read_data_size = (UInt64)3 * sizeof(Int32);
+            
+            using var attr_name = new MarshaledString("a1");
+            using var rows_name = new MarshaledString("rows");
+            using var cols_name = new MarshaledString("cols");
+            
+            tiledb_query_t* read_query;
+            Methods.tiledb_query_alloc(ctx, array, tiledb_query_type_t.TILEDB_READ, &read_query);
+            check_err(ctx);
+
+            byte[] row_start = Encoding.ASCII.GetBytes("a");
+            byte[] row_end = Encoding.ASCII.GetBytes("c");
+            var start_handle = GCHandle.Alloc(row_start, GCHandleType.Pinned);
+            var end_handle = GCHandle.Alloc(row_end, GCHandleType.Pinned);
+            Methods.tiledb_query_add_range_var_by_name(ctx, read_query, rows_name,
+            start_handle.AddrOfPinnedObject().ToPointer(), (ulong)row_start.Length,
+            end_handle.AddrOfPinnedObject().ToPointer(), (ulong)row_end.Length
+            );
+            check_err(ctx);
+
+            int cols_start = 2, cols_end = 4;
+            Methods.tiledb_query_add_range(ctx, read_query, 1, &cols_start, &cols_end, null);
+            check_err(ctx);
+
+            Methods.tiledb_query_set_layout(ctx, read_query, tiledb_layout_t.TILEDB_ROW_MAJOR);
+            check_err(ctx);
+
+            var read_data_handle = GCHandle.Alloc(read_data, GCHandleType.Pinned);
+            var read_data_size_handle = GCHandle.Alloc(read_data_size, GCHandleType.Pinned);
+            Methods.tiledb_query_set_data_buffer(ctx, read_query, attr_name,
+                read_data_handle.AddrOfPinnedObject().ToPointer(), (ulong*)read_data_size_handle.AddrOfPinnedObject()

```

Comment:
We need to use distinct variables/pointers for each of the buffer size handles here -- explanation below.